test/excluded_prefixes_collector_test.go

```diff
@@ -0,0 +1,337 @@
+	et.Require().NoError(err)
+
+	err = prefixPool.ReleaseExcludedPrefixes(kubeAdmPrefixes)
+	et.Require().NoError(err)
+
+	userPrefixes := prefixes{}
+	err = k8s_yaml.NewYAMLOrJSONDecoder(
+		strings.NewReader(userConfigMap.Data[prefixesFileName]), bufferSize,
+	).Decode(&userPrefixes)
+	et.Require().NoError(err)
+
+	err = prefixPool.ReleaseExcludedPrefixes(userPrefixes.Prefixes)
+	et.Require().NoError(err)
+
+	et.Eventually(et.checkPrefixes(prefixPool.GetPrefixes()), time.Second*15, time.Second)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithCorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"127.0.0.0/8",
+		"134.65.0.0/16",
+	}
+
+	prefixPool, _ := prefixpool.New(envPrefixes...)
+	et.deployCollectorWithEnvs(envPrefixes)
+
+	kubeAdmPrefixes, err := kubeAdmPrefixes()
+	et.Require().NoError(err)
+
+	err = prefixPool.ReleaseExcludedPrefixes(kubeAdmPrefixes)
+	et.Require().NoError(err)
+
+	et.Eventually(et.checkPrefixes(prefixPool.GetPrefixes()), time.Second*15, time.Second)
+}
+
+func (et *ExcludedPrefixesSuite) TestWithIncorrectEnvPrefixes() {
+	envPrefixes := []string{
+		"256.256.256.0",
+	}
+	et.deployCollectorWithEnvs(envPrefixes)
+
+	et.Eventually(func() bool {
+		podInfo, err := k8s.DescribePod(prefixesFileName, "", map[string]string{
+			"app": "excluded-prefixes-collector",
+		})
+		et.Require().NoError(err)
+		return podInfo == nil
+	}, time.Second*15, time.Second)
+}
+
+func TestExcludedPrefixesSuite(t *testing.T) {
+	suite.Run(t, &ExcludedPrefixesSuite{})
+}
+
+func (et *ExcludedPrefixesSuite) deployCollector() {
+	et.Require().NoError(exechelper.Run("kubectl apply -f ../deployments/prefixes-collector/collector.yaml", et.options...))
+	et.waitForPodStart(collectorNamespace, "excluded-prefixes-collector")
+	logrus.Info("Collector deployed")
+}
+
+func (et *ExcludedPrefixesSuite) deployCollectorWithEnvs(envPrefixes []string) {
+	et.Require().NoError(k8s.ApplyDaemonSet("../deployments/prefixes-collector/collector.yaml", func(collector *v1.DaemonSet) {
+		collector.Spec.Template.Spec.Containers[0].Env = append(collector.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  excludedPrefixesEnv,
+			Value: strings.Join(envPrefixes, ","),
+		})
+	}))
+	et.waitForPodStart(collectorNamespace, "excluded-prefixes-collector")
+	logrus.Info("Collector deployed")
+}
+
+func (et *ExcludedPrefixesSuite) checkPrefixes(expectedPrefixes []string) func() bool {
+	dirName := os.TempDir()
+	kubeCpCmd := fmt.Sprintf("kubectl cp -n %v %v:%v %v", collectorNamespace, et.alpinePodName, nsmConfigDir, dirName)
+	localPrefixesPath := filepath.Join(dirName, prefixesFileName)
+	return func() bool {
+		et.Require().NoError(exechelper.Run(kubeCpCmd))
+		actualPrefixes, err := prefixesFromFile(localPrefixesPath)
+		logrus.Infof("Actual: %v Expected: %v", actualPrefixes, expectedPrefixes)
+		return err == nil && reflect.DeepEqual(expectedPrefixes, actualPrefixes)
+	}
+}
+
+func (et *ExcludedPrefixesSuite) waitForPodStart(namespace, appLabelValue string) {
```

Comment:
Please do not use waits. If you need to check some state you can consider to use `testify.Eventually`

Reply:
Waits were removed.